File: backend/src/main/java/com/techgarage/service/JobService.java
package com.techgarage.service;

import com.techgarage.dto.job.*;
import java.util.List;
import com.techgarage.dto.job.ChangeRequestCreate;
import com.techgarage.dto.job.ChangeRequestResponse;

public interface JobService {
    List<JobResponse> getMyJobs();
    JobResponse getById(Long id);
    JobResponse updateStatus(Long id, JobStatusUpdateRequest request);
    JobResponse submitSolution(Long id, SolutionSubmitRequest request);
    JobResponse requestRevision(Long id, RevisionRequest request);
    JobResponse complete(Long id);
    JobResponse cancel(Long id);
    List<ChangeRequestResponse> getChangeRequests(Long jobId);
    ChangeRequestResponse createChangeRequest(Long jobId, ChangeRequestCreate request);
    ChangeRequestResponse respondToChangeRequest(Long id, boolean accept);
}

File: TechGarage-clean/frontend/src/services/jobService.js
import api, { unwrap } from './api'

export const jobService = {
  getMine: () => unwrap(api.get('/jobs')),
  getById: (id) => unwrap(api.get(`/jobs/${id}`)),
  updateStatus: (id, payload) => unwrap(api.put(`/jobs/${id}/status`, payload)),
  submitSolution: (id, payload) => unwrap(api.post(`/jobs/${id}/submit-solution`, payload)),
  requestRevision: (id, payload) => unwrap(api.post(`/jobs/${id}/revision`, payload)),
  complete: (id) => unwrap(api.post(`/jobs/${id}/complete`)),
  cancel: (id) => unwrap(api.post(`/jobs/${id}/cancel`)),
  raiseDispute: (id, payload) => unwrap(api.post(`/jobs/${id}/disputes`, payload)),
  getChangeRequests: (id) => unwrap(api.get(`/jobs/${id}/change-requests`)),
  createChangeRequest: (id, payload) => unwrap(api.post(`/jobs/${id}/change-requests`, payload)),
  respondToChangeRequest: (changeRequestId, accept) => unwrap(api.put(`/jobs/change-requests/${changeRequestId}/respond`, null, { params: { accept } })),
}
